tests/src/core/testqgsexpression.cpp

```diff
@@ -2875,6 +2875,15 @@ class TestQgsExpression: public QObject
       QTest::newRow( "translate" ) << "translate( $geometry, -1, 2)" << geom << false << true << QgsGeometry::fromWkt( QStringLiteral( "LINESTRING (-1 2, 9 12)" ) );
       geom = QgsGeometry::fromPointXY( point );
       QTest::newRow( "translate" ) << "translate( $geometry, 1, -2)" << geom << false << true << QgsGeometry::fromWkt( QStringLiteral( "POINT(1 -2)" ) );
+
+      geom = QgsGeometry::fromPolygonXY( polygon );
+      QTest::newRow( "rotate" ) << "rotate( $geometry, 90, geomFromWKT( 'POINT(10 10)' ) )" << geom << false << true << QgsGeometry::fromWkt( QStringLiteral( "Polygon ((0 20, 10 10, 0 10, 0 20))" ) );
+      geom = QgsGeometry::fromPolylineXY( line );
+      QTest::newRow( "rotate" ) << "rotate( $geometry, -90, geomFromWKT( 'POINT(5 5)' ) )" << geom << false << true << QgsGeometry::fromWkt( QStringLiteral( "LineString (10 0, 0 10)'" ) );
+      geom = QgsGeometry::fromPointXY( point );
+      QTest::newRow( "rotate" ) << "rotate( $geometry, 180, geomFromWKT( 'POINT(-5 -3)' ) )" << geom << false << true << QgsGeometry::fromWkt( QStringLiteral( "Point (-10 -6)" ) );
+      QTest::newRow( "rotate" ) << "rotate( $geometry, 180, geomFromWKT( 'MULTIPOINT((-5 -3))' ) )" << geom << false << true << QgsGeometry::fromWkt( QStringLiteral( "Point (-10 -6)" ) );
+      QTest::newRow( "rotate" ) << "rotate( $geometry, 180, geomFromWKT( 'MULTIPOINT((-5 -3, 7 7, 4 4))' ) )" << geom << true << true << QgsVariant();
```

Comment:
```suggestion
      QTest::newRow( "rotate" ) << "rotate( $geometry, 180, geomFromWKT( 'MULTIPOINT((-5 -3, 7 7, 4 4))' ) )" << geom << true << true << QVariant();
```

Reply:
Tnx! I tried but still get an error cause the test seems to expect a geometry:

QDEBUG : TestQgsExpression::eval_geometry_method() expected data of type 'QgsGeometry', got 'QVariant' for element 4 of data with tag 'rotate'
QFATAL : TestQgsExpression::eval_geometry_method() ASSERT: "false" in file qtestdata.cpp, line 92
FAIL!  : TestQgsExpression::eval_geometry_method() Received a fatal error.

Reply:
Try QgsGeometry() (an invalid geometry) instead of a QVariant() in this case